### ISSUE TITLE
variable with `Literal[False]` incorrectly marked as reachable

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#3639]
**Describe the bug**
if a variable with type `Literal[False]` is checked in an `if` statement, it's incorrectly marked as reachable

**Expected behavior**
code within the `if` statement is marked as unreachable

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/57028336/176593845-7ea05fb7-0cf6-447d-8c2a-5082ac9f76ce.png)

```python
if False:
    print(1)  # pyright unreachable

foo: Literal[False] = False
if foo:
    print(1)  # not unreachable
```

**VS Code extension or command-line**
vscode

**Comment by erictraut** [comment]
This is "as designed" behavior. The "grayed-out" text in this case means that the type analyzer does not analyze this block of code because the binder (an earlier phase of analysis) has determined that it's unreachable. The binder runs prior to type analysis, and it understands only certain expression forms such as `False`, `True`, and `sys.version_info >= (3, 9)` — expression forms that are specified in PEP 484.

**Comment by erictraut** [comment]
@DetachHead, I noticed that whenever you file a bug, you "like" it, and then @KotlinIsland also "likes" it within a second or two. I presume that you are one person using two different github accounts? Why do you do this? Just curious.

**Comment by DetachHead** [comment]
he is my gang stalker
